src/components/App/App.js

```diff
@@ -50,16 +77,33 @@ class App extends Component {
   }
   
   render() {
+    console.log('SELECTED TREE AT THE ACTUAL END', this.state.selectedTree)
+    // const page = this.state.selectedZip && this.state.selectedTree ? '/details' : '/result' 
+    // page = selectedTree ? '/details' : '/result'
+    // console.log(page, "PAGE71")
     return (
       <>
-      <p>App Component</p>
-      <Header />
-      <Home changeZipCode={this.changeZipCode}/>
-      <SearchResult />
-      <DetailsContainer />
+        <Header clearZipTrees={this.clearZipTrees}/>
+        <Switch>
+          {/* <Route exact path='/' render={() => <Home changeZipCode={this.changeZipCode}/>}/> */}
+          <Route exact path="/">
+	          {this.state.selectedZip ? <Redirect to="/result" /> : <Home changeZipCode={this.changeZipCode}/>}
+          </Route>;
+          <Route exact path="/result">
+	          {this.state.selectedTree ? <Redirect to="/trees/:id" /> : <SearchResult filteredTrees={this.state.filteredTrees} changeSelectedTree={this.changeSelectedTree} clearZipTrees={this.clearZipTrees}/>}
+          </Route>; 
+
+          <Route exact path='/trees/:id' render={() => <DetailsContainer selectedTree={this.state.selectedTree} clearSelectedTree={this.clearSelectedTree}/>}/>
```

Comment:
This dynamic path isn't working the way I expect it.  I don't know if we need to change the path on this route.